### PR TITLE
Restructure and add description of e2e tests

### PR DESCRIPTION
**Pull request by aorcholski**
# Description

All tests moved to the `test/scenarios` folder.
All e2e tests documented in README.md files.

## Checklist
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: test/scenarios/cloudnative/basic/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+- sample application restarted
+- check if oneagent init container is injected
+- check if oneagents are able to communicate with the cluster
+
+### CodeModules
+Verify that the storage in the CSI driver directory does not increase when there are multiple tenants and pods which are monitored.
+
+- DTO installed
+- sample application Deployment installed
+- check if oneagent started
```
Comment (StefanHauth) [inline test/scenarios/cloudnative/basic/README.md:36]:
```suggestion
- check if OneAgent started
```

Comment (StefanHauth) [reply test/scenarios/cloudnative/basic/README.md:36]: AFAIK OneAgent is a kind of brand name or something so it should be written in capital letters. This comment applies to all uses of `oneagent` in this PR.

**Review by StefanHauth — APPROVED** [review]
I'm a bit doubtful of separating test code and test description. In my experience the latter one has a hard time getting maintained / is constantly overlooked when tests change. I'd prefer a description of the basic idea behind each test as a comment close to the test code. And it shouldn't be a step-by-step description of what the test does, rather describing the intentions of the test.
Other than that, one step further into a better future of the E2E tests.

File: test/scenarios/activegate/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Prerequisites
+k8s cluster
```
Comment (chrismuellner) [inline test/scenarios/activegate/README.md:2]: `k8s cluster` is implied, I think only additional prerequisites are required

File: test/scenarios/applicationmonitoring/appmon_test.go [hunk 3]
```diff
@@ -1,6 +1,6 @@
 //go:build e2e
 
-package appmon
+package applicationmonitoring
```
Comment (chrismuellner) [inline test/scenarios/applicationmonitoring/appmon_test.go:3]: file name should also be adjusted

File: test/scenarios/cloudnative/basic/README.md (hunk 1 above)
Comment (chrismuellner) [reply test/scenarios/cloudnative/basic/README.md:36]: not official documentation, so unnecessary in my opinion

File: test/scenarios/applicationmonitoring/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Prerequisites
+k8s cluster
+
+# Setup
+ApplicationMonitoring deployment without CSI driver
+
+# Goals
+### Verification of the data ingest part of the operator
```
Comment (chrismuellner) [inline test/scenarios/applicationmonitoring/README.md:8]: wrong section level

File: test/scenarios/activegate/README.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+# Setup
+CloudNative deployment with CSI driver
```
Comment (chrismuellner) [inline test/scenarios/activegate/README.md:5]: this seems incorrect

File: test/scenarios/activegate/README.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+# Goals
+Verification if ActiveGate is rolled out successfully. All AG capabilities are enabled in Dynakube.
+
+Tests:
```
Comment (chrismuellner) [inline test/scenarios/activegate/README.md:10]: we should split this into a numbered list of steps and then a sub list with a rough outline of checks done for each

example:
1. create Activegate
  - verify pod is running
  - ...
2. ...

**Review by chrismuellner — CHANGES_REQUESTED** [review]

**Comment by luhi-DT** [comment]
> I'm a bit doubtful of separating test code and test description. In my experience the latter one has a hard time getting maintained / is constantly overlooked when tests change. I'd prefer a description of the basic idea behind each test as a comment close to the test code. And it shouldn't be a step-by-step description of what the test does, rather describing the intentions of the test. Other than that, one step further into a better future of the E2E tests.

I also think that the readme's are not a good idea here, what we should have is a small seperate documentation like a wiki page and the rest (details etc.) should be taken from the code

**Review by chrismuellner — APPROVED** [review]

**Pull request merged**